### PR TITLE
Seal Recovery: sys/init-unseal-recovery and documentation

### PR DESCRIPTION
**Pull request by sgmiller**
Adds the sys endpoint for establishing unseal recovery after the fact, for
pre 1.13 clusters or clusters initialized with disable_unseal_recovery.

Docs for unseal recovery in general.

### CODE REVIEW

File: http/handler.go [hunk 1]
```diff
@@ -172,6 +172,7 @@ func handler(props *vault.HandlerProperties) http.Handler {
 		mux.Handle("/v1/sys/rekey-recovery-key/init", handleRequestForwarding(core, handleSysRekeyInit(core, true)))
 		mux.Handle("/v1/sys/rekey-recovery-key/update", handleRequestForwarding(core, handleSysRekeyUpdate(core, true)))
 		mux.Handle("/v1/sys/rekey-recovery-key/verify", handleRequestForwarding(core, handleSysRekeyVerify(core, true)))
+		mux.Handle("/v1/sys/init-unseal-recovery", handleRequestForwarding(core, handleSysInitUnsealRecovery(core)))
```
Comment (ncabatoff) [inline http/handler.go:175]:
```suggestion
		mux.Handle("/v1/sys/enable-unseal-recovery", handleRequestForwarding(core, handleSysInitUnsealRecovery(core)))
```

I think the associations of your proposed endpoint with sys/init make sense on one level, but are inappropriate on another, i.e. it implies that this must be done on an un-initialized cluster.

**Comment by ncabatoff** [comment]
Is there a way for an operator to identify whether their cluster supports unsealing with recovery keys, i.e. whether this endpoint needs invoking?  Can we add one if not, e.g. a new field in sys/seal-status that appears in `vault status`?

File: website/content/api-docs/system/enable-unseal-recovery.mdx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+---
+
+# `/sys/init-unseal-recovery`
+
+The `/sys/init-unseal-recovery` endpoint is used to enable the unsealing of
+Vault via recovery keys for clusters that were initialized prior to version 1.13
+or which were initialized with `disable_unseal_recovery` true.
+This feature can only be used if using an automatic unseal mechanism with recovery
+keys.  Once successful, auto-seals can be placed in 
+[unseal recovery mode](/docs/concepts/seal#unseal-recovery-mode).
```
Comment (kendallstrautman) [inline website/content/api-docs/system/enable-unseal-recovery.mdx:14]:
```suggestion
[unseal recovery mode](/vault/docs/concepts/seal#unseal-recovery-mode).
```
Hello 👋🏻 Now that the [link migration work](https://github.com/hashicorp/vault/pull/18696) has been merged, I'm visiting open PRs to resolve legacy links. All links should now use the full path from the root of developer.hashicorp.com. Let me know if you have any questions!

File: website/content/docs/concepts/seal.mdx [hunk 3]
```diff
@@ -172,6 +173,17 @@ endpoint](/api-docs/system/rekey); however, the
...
 
+## Unseal Recovery Mode
+
+Auto seals that have failed or whose keys have been lost may in an emergency 
+be unsealed using the recovery keys by placing the seal in Unseal Recovery Mode.
+To do this, edit the seal stanza for the auto-seal, adding the option 
+`recovery_keys_fallback = "true"`.  Then, using the recovery keys, unseal vault 
+as if it was a Shamir seal.  This is possible if the cluster was initialized in 
+version 1.13 or above, and `unseal-recovery-disabled` was not true at the time 
+of initialization, _or_ [/sys/init-unseal-recovery](/api-docs/system/init-unseal-recovery)
```
Comment (kendallstrautman) [inline website/content/docs/concepts/seal.mdx:184]:
```suggestion
of initialization, _or_ [/sys/init-unseal-recovery](/vault/api-docs/system/init-unseal-recovery) 
```

**Comment by sgmiller** [comment]
> Is there a way for an operator to identify whether their cluster supports unsealing with recovery keys, i.e. whether this endpoint needs invoking? Can we add one if not, e.g. a new field in sys/seal-status that appears in `vault status`?

Great idea.

**Comment by sgmiller** [comment]
Holding this in draft to switch to a rekey based approach.

**Comment by sgmiller** [comment]
Switched to a new branch for this work

File: website/content/docs/concepts/seal.mdx [hunk 4]
```diff
@@ -301,3 +316,4 @@ When the quorum of nodes are back up, Raft will elect a leader and the leader
 node that will perform the migration. The migrated information will be replicated to
 all other cluster peers and when the peers eventually become the leader,
 migration will not happen again on the peer nodes.
+r
```
Comment (ncabatoff) [inline website/content/docs/concepts/seal.mdx:319]: Typo it looks like.

**Pull request closed without merging**